contracts/external/policies/TimeFramePolicy.sol

```diff
@@ -0,0 +1,123 @@
+
+    mapping(ConfigId id => mapping(address msgSender => mapping(address opSender => TimeFrameConfig))) public
+        timeFrameConfigs;
+
+    /**
+     * @notice Checks if the user operation is within the valid time frame.
+     * @param id The config ID.
+     * @param op The user operation.
+     */
+    function checkUserOpPolicy(ConfigId id, PackedUserOperation calldata op) external view override returns (uint256) {
+        return _checkTimeFrame(id, msg.sender, op.sender);
+    }
+
+    /**
+     * @notice Checks if the action is within the valid time frame.
+     * @param id The config ID.
+     * @param account The account.
+     */
+    function checkAction(
+        ConfigId id,
+        address account,
+        address,
+        uint256,
+        bytes calldata
+    )
+        external
+        view
+        returns (uint256)
+    {
+        return _checkTimeFrame(id, msg.sender, account);
+    }
+
+    /**
+     * @notice Checks if the 1271 signed action is within the valid time frame.
+     * @param id The config ID.
+     * @param smartAccount The account.
+     */
+    function check1271SignedAction(
+        ConfigId id,
+        address,
+        address smartAccount,
+        bytes32,
+        bytes calldata
+    )
+        external
+        view
+        returns (bool)
+    {
+        TimeFrameConfig memory config = timeFrameConfigs[id][msg.sender][smartAccount];
+        require(config.validUntil != 0 || config.validAfter != 0, PolicyNotInitialized(id, msg.sender, smartAccount));
+        if ((block.timestamp < config.validUntil || config.validUntil == 0) && block.timestamp >= config.validAfter) {
+            return true;
+        }
+        return false;
+    }
+
+    function _checkTimeFrame(ConfigId id, address multiplexer, address smartAccount) internal view returns (uint256) {
+        TimeFrameConfig memory config = timeFrameConfigs[id][multiplexer][smartAccount];
+        require(config.validUntil != 0 || config.validAfter != 0, PolicyNotInitialized(id, multiplexer, smartAccount));
+        return _packValidationData({sigFailed:false, validUntil: config.validUntil, validAfter: config.validAfter});
+    }
+
+    /**
+     * @notice Initializes the policy. 
+     * Overwrites state.
+     * @notice ATTENTION: This method is called during permission installation as part of the enabling policies flow.
+     * A secure policy would minimize external calls from this method (ideally, to 0) to prevent passing control flow to
+     * external contracts.
+     * @param account The account.
+     * @param configId The config ID.
+     * @param initData The initialization data.
+     */
+    function initializeWithMultiplexer(address account, ConfigId configId, bytes calldata initData) external {
+        timeFrameConfigs[configId][msg.sender][account].validUntil = uint48(uint128(bytes16(initData[0:16])));
+        timeFrameConfigs[configId][msg.sender][account].validAfter = uint48(uint128(bytes16(initData[16:32])));
```

Comment:
this encoding is kind of weird - why dont we just pack two uint48s together, the amount of zeros is the same anyways

Reply:
addressed here https://github.com/erc7579/smartsessions/pull/122/commits/b6701781dfac7175549caeba9f6197d9b058ee7b